packages/jest-worker/src/base/BaseWorkerPool.ts

```diff
@@ -85,15 +90,32 @@ export default class BaseWorkerPool {
     throw Error('Missing method createWorker in WorkerPool');
   }
 
-  end(): void {
+  async end(): Promise<PoolExitResult> {
     // We do not cache the request object here. If so, it would only be only
     // processed by one of the workers, and we want them all to close.
-    for (let i = 0; i < this._workers.length; i++) {
-      this._workers[i].send(
-        [CHILD_MESSAGE_END, false],
-        emptyMethod,
-        emptyMethod,
-      );
-    }
+    const workerExitPromises = this._workers.map(async worker => {
+      worker.send([CHILD_MESSAGE_END, false], emptyMethod, emptyMethod);
+
+      // Schedule a force exit in case worker fails to exit gracefully so
+      // await worker.waitForExit() never takes longer than FORCE_EXIT_DELAY
+      let forceExited = false;
+      const forceExitTimeout = setTimeout(() => {
+        worker.send([CHILD_MESSAGE_FORCE_EXIT], emptyMethod, emptyMethod);
```

Comment:
~Shower~Bed thought: Actually, this is obviously not gonna work. `END` just stopped the listener on the exact same channel that I'm now trying to send the force exit request through 🤦‍♂️
It does detect the leak and result in `{forceExited:true}`, which is why I didn't notice the mistake, but I don't think the worker will die.
Should perhaps `SIGTERM` from the outside instead?

Reply:
`ChildProcessWorker` now sends `SIGTERM` for force exiting, and even `SIGKILL` after further delay